src/app/api/analytics/analytics.py

```diff
@@ -91,27 +91,42 @@ class CollectionValidationStats(ElasticValidationStats[list[OehValidationError]]
     pass
 
 
-def none_to_empty_list(v: list) -> Optional[list]:
-    if v is None:
-        return []
-    return v
+class MaterialValidationStats(BaseModel):
+    """
+    The UUIDs within the optional lists identify materials.
+    """
+
+    subjects: Optional[list[uuid.UUID]]
+    license: Optional[list[uuid.UUID]]
+    url: Optional[list[uuid.UUID]]
+    publisher: Optional[list[uuid.UUID]]
+    intended_end_user_role: Optional[list[uuid.UUID]]
+    material_type: Optional[list[uuid.UUID]]
+    title: Optional[list[uuid.UUID]]
+    keywords: Optional[list[uuid.UUID]]
+    description: Optional[list[uuid.UUID]]
+    edu_context: Optional[list[uuid.UUID]]
 
 
-class MaterialFieldValidation(BaseModel):
-    missing: Optional[list[uuid.UUID]]
-    too_short: Optional[list[uuid.UUID]]
-    too_few: Optional[list[uuid.UUID]]
-    lacks_clarity: Optional[list[uuid.UUID]]
-    invalid_spelling: Optional[list[uuid.UUID]]
+class MaterialValidationResponse(BaseModel):
+    collection_id: uuid.UUID
+    derived_at: datetime = Field(default_factory=datetime.now)
+    stats: MaterialValidationStats
+
 
-    # validators
-    _none_to_empty_list = validator("*", pre=True, allow_reuse=True)(none_to_empty_list)
+class PendingMaterials(BaseModel):
+    node_id: uuid.UUID
```

Comment:
is this node_id the same as the collection_id in the outer PendingMaterialsResponse?
what is it the id of? I assume a collection!

Reply:
It is a collection id - changed it.